src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,92 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationService REGISTRATION_SERVICE = new RegistrationServiceImpl();
+
+    @Test
+    void register_theSameLogin_NotOk() {
+        User testUser = new User();
+        testUser.setLogin("WormJim");
+        testUser.setPassword("123456");
+        testUser.setAge(29);
+        User testUser2 = new User();
+        testUser2.setLogin("WormJim");
+        testUser2.setPassword("654321");
+        testUser2.setAge(24);
+        User expected = REGISTRATION_SERVICE.register(testUser);
+        User actual = REGISTRATION_SERVICE.register(testUser2);
+        assertEquals(expected, actual, "Test failed you should not register the same user's login");
+    }
+
+    @Test
+    void register_invalidLoginLength_notOk() {
+        User testUser = new User();
+        User testUser2 = new User();
+        testUser.setLogin("Jo");
+        testUser.setPassword("123456");
+        testUser2.setLogin("");
+        testUser2.setPassword("123456");
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser2);
+        });
+    }
```

Comment:
Good job on testing for invalid login lengths. However, remember to also test for the edge cases as mentioned in the checklist. For example, a login length of 1 (not-ok edge case), a login length of 2 (ok edge case), and a login length of 3 (ok).